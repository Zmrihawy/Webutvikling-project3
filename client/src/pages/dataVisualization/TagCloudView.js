import React, { Component } from "react";
import randomColor from "randomcolor";
import TagCloud from "react-tag-cloud";
import CloudItem from "./CloudItem";

import "./tag-cloud-style.css";
class TagCloudView extends Component {
  componentDidMount() {
    setInterval(() => {
      this.forceUpdate();
    }, 3000);
  }

  render() {
    // componentStatistics is an object where each key is the name of a statistics and each value is the number
    // of objects in the db by that statistics
    const { statistics } = this.props;
    let mappedProjectedVals = [];
    if (statistics) {
      const statisticsValues = Object.keys(statistics).map(
        statistic => statistics[statistic]
      );
      const oldMax = Math.max(...statisticsValues);
      const oldMin = Math.min(...statisticsValues);
      const oldRange = oldMax - oldMin;
      const newMin = 10;
      const newMax = 100;
      const newRange = newMax - newMin;
      // Create a new array of objects, where each object contains the name of the statistics and the projected count value
      const projectedVals = Object.keys(statistics).map(statistic => {
        let oldVal = statistics[statistic];
        let newVal = ((oldVal - oldMin) * newRange) / oldRange + newMin;
        return { statistic, value: newVal };
      });
      // Map to JSX
      mappedProjectedVals = projectedVals.map((object, i) => (
        <div key={i} style={{ fontSize: object.value }}>
          {object.statistic}
        </div>
      ));
    }

    return (
      <div className="app-outer">
        <div className="app-inner">
          <TagCloud
            className="tag-cloud"
            style={{
              fontFamily: "sans-serif",
              fontSize: 30,
              color: () => randomColor(),
              padding: 5
            }}
          >
            {mappedProjectedVals}
          </TagCloud>
        </div>
      </div>
    );
  }
}

export default TagCloudView;
