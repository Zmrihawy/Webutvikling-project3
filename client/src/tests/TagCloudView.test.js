import React from "react";
import TagCloudView from "../pages/dataVisualization/TagCloudView"
import renderer from "react-test-renderer";

it("renders tag cloud view", () => {
  let stats = {};
  stats.stat1 = 10;
  stats.stat2 = 137;
  stats.stat3 = 1;
  const tree = renderer.
    create(<TagCloudView statistics={stats} />)
  expect(tree).toMatchSnapshot();
});

