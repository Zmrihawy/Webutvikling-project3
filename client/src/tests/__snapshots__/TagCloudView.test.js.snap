// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tag cloud view 1`] = `
<div
  className="app-outer"
>
  <div
    className="app-inner"
  >
    <div
      className="tag-cloud"
      style={Object {}}
    />
  </div>
</div>
`;
